File: app/src/main/java/com/velaphi/untamed/features/animalDetails/models/Info.java
package com.velaphi.untamed.features.animalDetails.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.firestore.IgnoreExtraProperties;

@IgnoreExtraProperties
public class Info implements Parcelable {

    private String title;
    private String description;
    public static final Creator<Info> CREATOR = new Creator<Info>() {
        @Override
        public Info createFromParcel(Parcel in) {
            return new Info(in);
        }

        @Override
        public Info[] newArray(int size) {
            return new Info[size];
        }
    };

    public Info() {
    }

    private int order;


    public Info(String title, String description, int order) {
        this.title = title;
        this.description = description;
        this.order = order;
    }

    protected Info(Parcel in) {
        title = in.readString();
        description = in.readString();
        order = in.readInt();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(title);
        dest.writeString(description);
        dest.writeInt(order);
    }

    public static Creator<Info> getCREATOR() {
        return CREATOR;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public int getOrder() {
        return order;
    }


    @Override
    public int describeContents() {
        return 0;
    }

}

File: app/src/main/java/com/velaphi/untamed/features/animalDetails/models/Info.kt
package com.velaphi.untamed.features.animalDetails.models

import android.os.Parcel
import android.os.Parcelable
import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
class Info : Parcelable {
    var title: String? = null
        private set
    var description: String? = null
        private set

    constructor() {}

    var order = 0
        private set

    constructor(title: String?, description: String?, order: Int) {
        this.title = title
        this.description = description
        this.order = order
    }

    protected constructor(`in`: Parcel) {
        title = `in`.readString()
        description = `in`.readString()
        order = `in`.readInt()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(title)
        dest.writeString(description)
        dest.writeInt(order)
    }

    override fun describeContents(): Int {
        return 0
    }


    companion object CREATOR : Parcelable.Creator<Info> {
        override fun createFromParcel(parcel: Parcel): Info {
            return Info(parcel)
        }

        override fun newArray(size: Int): Array<Info?> {
            return arrayOfNulls(size)
        }
    }
}